File: app/src/main/java/com/sk/revisit/data/UrlLog.java
package com.sk.revisit.data;

public class UrlLog {
	public String url;
	public long size;
	double p;
	boolean isComplete;
	private OnProgressChangeListener listener;

	UrlLog(String urlText, long size) {
		this.url = urlText;
		this.size = size;
		this.isComplete = false;
	}

	public void setIsComplete(boolean o) {
		this.isComplete = o;
	}

	public void setProgress(double p) {
		this.p = p;
		listener.onChange(p);
	}

	public void setOnProgressChangeListener(OnProgressChangeListener listener) {
		this.listener = listener;
	}

	public interface OnProgressChangeListener {
		void onChange(double p);
	}
}

File: app/src/main/java/com/sk/revisit/data/UrlLog.kt
package com.sk.revisit.data

class UrlLog(val url: String, val size: Long) {
    var p: Double = 0.0
    var isComplete: Boolean = false
    private var listener: OnProgressChangeListener? = null

    fun setProgress(p: Double) {
        this.p = p
        listener?.onChange(p)
    }

    fun setOnProgressChangeListener(listener: OnProgressChangeListener?) {
        this.listener = listener
    }

    fun interface OnProgressChangeListener {
        fun onChange(p: Double)
    }
}
